plugin/helpers.go

```diff
@@ -163,49 +153,27 @@ func ToFileMode(mode interface{}) (os.FileMode, error) {
 	return fileMode, nil
 }
 
-// Attr returns the entry's attributes in a form that's consummable by
-// FUSE and the API. This may or may not match entry.Attr(). Specifically,
-//   * If entry.Attr() sets the size field to SizeUnknown and the entry supports
-//   the Read action, then the returned attributes' size will be set to
-//   the size of the entry's content as returned by Open.
-//
-//   * If entry.Attr() does not set the mode, then the returned attributes'
-//   mode will be set to "os.ModeDir | 0550" if the entry supports the List
-//   action; otherwise, it will be set to 0440. This is needed by FUSE.
-//
-//   * Otherwise, the returned attributes will match what's returned by
-//   entry.Attr().
-//
-func Attr(ctx context.Context, entry Entry) (Attributes, error) {
-	attr, err := entry.Attr(ctx)
-	if err != nil {
-		return attr, err
-	}
+// Attributes returns the entry's attribtues
+func Attributes(e Entry) EntryAttributes {
+	return e.attributes()
+}
 
-	if attr.Size == SizeUnknown && ReadAction.IsSupportedOn(entry) {
-		content, openErr := CachedOpen(ctx, entry.(Readable))
-		if openErr != nil {
-			err = ErrCouldNotDetermineSizeAttr{openErr.Error()}
-			attr.Size = 0
-		} else {
-			size := content.Size()
-			if size < 0 {
-				return attr, ErrNegativeSizeAttr{size}
-			}
-
-			attr.Size = uint64(size)
-		}
+// RefreshAttributes refreshes the entry's attributes. It is primarily
+// needed by FUSE because FUSE nodes have a very long lifetime.
+func RefreshAttributes(ctx context.Context, e Entry) error {
+	journal.Record(ctx, "Refreshing the attributes of %v", Path(e))
+
+	meta, err := CachedMetadata(ctx, e)
```

Comment:
Can't put this logic in EntryBase b/c it will call EntryBase's Metadata method.

Reply:
Seems like a useful code comment.